src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,156 @@
+
+            final Storage storage = storageDbRepository.getStorage(storageId)
+                    .orElseThrow(() -> new UnableProcessException("No storage with id: " + storageId));
+            final Timeline activeTimeline = getTimeline(eventType);
+            final TopicRepository currentTopicRepo =
+                    topicRepositoryHolder.getTopicRepository(activeTimeline.getStorage());
+            final TopicRepository nextTopicRepo = topicRepositoryHolder.getTopicRepository(storage);
+            final List<PartitionStatistics> partitionStatistics =
+                    currentTopicRepo.loadTopicStatistics(Collections.singleton(activeTimeline.getTopic()));
+
+            if (activeTimeline.isFake()) {
+                LOG.info("Switching from fake timeline to real one. Current topic {}", activeTimeline.getTopic());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, activeTimeline.getTopic(), new Date());
+                switchTimeline(eventType, nextTimeline, () -> nextTimeline.setSwitchedAt(new Date()));
+            } else {
+                final String newTopic = nextTopicRepo.createTopic(partitionStatistics.size(),
+                        eventType.getOptions().getRetentionTime());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, newTopic, new Date());
+                LOG.info("Switching timelines. Topics: {} -> {}", activeTimeline.getTopic(), nextTimeline.getTopic());
+                switchTimeline(eventType, nextTimeline, () -> {
+                    final Timeline.StoragePosition storagePosition =
+                            StoragePositionFactory.createStoragePosition(activeTimeline, currentTopicRepo);
+                    activeTimeline.setLatestPosition(storagePosition);
+                    nextTimeline.setSwitchedAt(new Date());
+                    timelineDbRepository.updateTimelime(activeTimeline);
+                });
+            }
+        } catch (final NakadiException ne) {
+            throw new TimelineException(ne.getMessage(), ne);
+        }
+    }
+
+    public Timeline getTimeline(final EventType eventType) throws TimelineException {
+        try {
+            final String eventTypeName = eventType.getName();
+            final Optional<Timeline> activeTimeline = eventTypeCache.getActiveTimeline(eventTypeName);
+            if (activeTimeline.isPresent()) {
+                return activeTimeline.get();
+            }
+
+            final Storage storage = storageDbRepository.getStorage(DEFAULT_STORAGE)
+                    .orElseThrow(() -> new UnableProcessException("Fake timeline creation failed for event type " +
+                            eventType.getName() + ".No default storage defined"));
```

Comment:
If I understood correctly, Nakadi no longer works without this `DEFAULT_STORAGE` in the database. That means that creating the `DEFAULT_STORAGE` in the database is an additional step for deploying Nakadi successfully.

Considering someone else uses Nakadi as opensource software, this should be documented in the release process otherwise it's just going to break in a unexpected manner.

Why not create this "default storage" directly from application configuration? Then Nakadi would have 2 modes of operation: single timeline (or default) and with multiple timelines.

It would make the release process simple and not force every Nakadi appliance user to run some "curl" during setup.